src/rust/ide/src/model/suggestion_database.rs

```diff
@@ -81,12 +81,14 @@ impl Entry {
     }
 
     /// Returns the code which should be inserted to Searcher input when suggestion is picked.
-    pub fn code_to_insert(&self) -> String {
+    pub fn code_to_insert(&self, this_var:Option<&str>) -> String {
         let module = self.module.name();
         if self.self_type.as_ref().contains(&module) {
-            format!("{}.{}",module,self.name)
-        } else if self.self_type.as_ref().contains(&constants::keywords::HERE) {
+            format!("{}.{}",this_var.unwrap_or(module),self.name)
+        } else if self.self_type.as_ref().contains(&constants::keywords::HERE) { // TODO when this happens? why would the *type* be "here"?
```

Comment:
My bad, the matter of `here` in suggestions should be discussed in an engine's issue somewhere, I'll do that and refer the issue here in my next PR.